include/ht_resource.h

```diff
@@ -6,6 +6,9 @@
 #include <ht_file.h>
 #include <map>
 #include <type_traits>
+#include <ht_path_singleton.h>
```

Comment:
Why are these additional includes in a file that actually doesn't need them?  We should only put these in the files that actually utilize these headers.